src/hooks/useSeries.ts

```diff
@@ -1,29 +1,35 @@
 import { useQuery, UseQueryResult } from 'react-query';
 
-import { getSeries, getMediaByIds } from '#src/services/api.service';
-import { enrichMediaItems } from '#src/utils/series';
-import type { Season, Series } from '#types/series';
+import { getMediaByIds, getSeries } from '#src/services/api.service';
+import { enrichMediaItems, getSeriesEpisodes } from '#src/utils/series';
+import type { Series } from '#types/series';
 import type { Playlist } from '#types/playlist';
 import type { ApiError } from '#src/utils/api';
 
-export default (seriesId: string): UseQueryResult<{ series: Series; playlist: Playlist }, ApiError> => {
+export default (seriesId?: string): UseQueryResult<{ series: Series; playlist: Playlist }, ApiError> => {
   return useQuery(
-    `series-${seriesId}`,
+    ['series', seriesId],
     async () => {
-      const series = await getSeries(seriesId);
-      const mediaIds = (series?.seasons || []).flatMap((season: Season) => season.episodes.map((episode) => episode.media_id));
+      const series = await getSeries(seriesId || '');
+      const mediaIds = series ? getSeriesEpisodes(series).map((episode) => episode.media_id) : [];
       const mediaItems = await getMediaByIds(mediaIds);
 
       return {
         series,
-        playlist: { title: series?.title, description: series?.description, feedid: series?.series_id, playlist: enrichMediaItems(series, mediaItems) },
+        playlist: {
+          title: series?.title,
+          description: series?.description,
+          feedid: series?.series_id,
+          playlist: enrichMediaItems(series, mediaItems),
+        },
       };
     },
     {
+      enabled: !!seriesId,
       // 8 hours
       staleTime: 60 * 1000 * 60 * 8,
-      retry: 2,
-      retryDelay: 200,
+      // don't retry when we got a not found (prevent unneeded requests)
+      retry: (failureCount, error) => error.code !== 404 && failureCount < 2,
```

Comment:
In theory we can get a 404 from a network hiccup and in that case we still DO want to retry, no?

Reply:
No, when a request fails due to network issues, it should never result in a 404 because 404 is the status returned from the response which didn't succeed

Reply:
I'm assuming we'd get a 404 at the content router level if a hiccup caused the service to be unreachable, but maybe that returns 400 or 401.  In any case, it can always be fixed by the user refreshing the page, so you're probably right that it's not worth the overhead.